apps/api-extractor/src/api/IConfigFile.ts

```diff
@@ -0,0 +1,365 @@
+
+  /**
+   * The root folder for the project.  This folder typically contains the tsconfig.json and package.json
+   * config files.
+   *
+   * @remarks
+   *
+   * The `rootFolder` path is resolved relative to the folder containing api-extractor.json.
+   *
+   * The default value for `rootFolder` is the token `<lookup>`, which means the folder is determined by traversing
+   * parent folders, starting from the folder containing api-extractor.json, and stopping at the first folder
+   * that contains a tsconfig.json file.  If a tsconfig.json file cannot be found in this way, then an error
+   * will be reported.
+   */
+  rootFolder: string;
+
+  /**
+   * Provides already parsed tsconfig.json contents.
+   *
+   * @remarks
+   * The value must conform to the TypeScript tsconfig schema:
+   *
+   * http://json.schemastore.org/tsconfig
+   *
+   * If omitted, then the tsconfig.json file will instead be read from the rootFolder.
+   */
+  overrideTsconfig?: { };
+
+  /**
+   * This option causes the compiler to be invoked with the `--skipLibCheck` option.
+   *
+   * @remarks
+   * This option is not recommended and may cause API Extractor to produce incomplete or incorrect declarations,
+   * but it may be required when dependencies contain declarations that are incompatible with the TypeScript engine
+   * that API Extractor uses for its analysis.  Where possible, the underlying issue should be fixed rather than
+   * relying on skipLibCheck.
+   */
+  skipLibCheck?: boolean;
+}
+
+/**
+ * Configures how the API review files (*.api.md) will be generated.
+ *
+ * @remarks
+ * This is part of the {@link IConfigFile} structure.
+ *
+ * @public
+ */
+export interface IConfigApiReport {
+  /**
+   * Whether to generate an API report.
+   */
+  enabled: boolean;
+
+  /**
+   * The filename for the API report files.  It will be combined with `reportFolder` or `reportTempFolder` to produce
+   * a full output filename.
+   *
+   * @remarks
+   * The file extension should be ".api.md", and the string should not contain a path separator such as `\` or `/`.
+   */
+  reportFileName?: string;
+
+  /**
+   * Specifies the folder where the API report file is written.  The file name portion is determined by
+   * the "reportFileName" setting.
+   *
+   * @remarks
+   * The API report file is normally tracked by Git.  Changes to it can be used to trigger a branch policy,
+   * e.g. for an API review.
+   *
+   * The path is resolved relative to the `rootFolder` location.
+   */
+  reportFolder?: string;
+
+  /**
+   * Specifies the folder where the temporary report file is written.  The file name portion is determined by
+   * the "reportFileName" setting.
+   *
+   * @remarks
+   * After the temporary file is written to disk, it is compared with the file in the `reportFolder`.
+   * If they are different, a production build will fail.
+   *
+   * The path is resolved relative to the `rootFolder` location.
+   */
+  reportTempFolder?: string;
```

Comment:
I've always found this sorta annoying. Why not just write directly to the file in the `reportFolder` in non-production mode and read and compare in production mode?

Reply:
The error message says:

```
You have changed the public API signature for this project. 
Please overwrite ___ with a copy of ___ and then request an API review. 
See the Git repository README.md for more info.
```

We could change it to tell them to re-run API Extractor with the `--local` flag, but if it gets invoked as part of a complex build script, that might not be straightforward.